explorer/templates.go

```diff
@@ -303,6 +304,37 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 		"toLowerCase": func(a string) string {
 			return strings.ToLower(a)
 		},
+		"fetchListNavURL": func(groupingStr string, interval int64) string {
+			// This allows all links in years page to be linked to months i.e.
+			// /year -> /months -> weeks -> /days -> /blocks
+			var matchedGrouping string
+			val := dbtypes.TimeGroupingFromStr(groupingStr)
+
+			switch val {
+			case dbtypes.YearGrouping:
+				matchedGrouping = "months"
+			case dbtypes.MonthGrouping:
+				matchedGrouping = "weeks"
+			case dbtypes.WeekGrouping:
+				matchedGrouping = "days"
+			// for dbtypes.DayGrouping and any other groupings default to blocks.
+			default:
+				return fmt.Sprintf("/blocks?offset=%d", interval)
+			}
+			matchingVal := dbtypes.TimeGroupingFromStr(matchedGrouping)
+			intervalVal, err := dbtypes.TimeBasedGroupingToInterval(matchingVal)
+			if err != nil {
+				log.Errorf("Resolving the new group interval failed: error : %v", err)
+				return "/blocks?offset=0"
+			}
+
+			latestTime := time.Now().Unix()
+			offsetVal := (latestTime - interval) / int64(intervalVal)
+			if offsetVal < 0 {
+				return "/blocks?offset=0"
```

Comment:
Would you document the logic in this check and exception?  I guess the idea is to default to /blocks with zero offset if the passed `interval` is in the future?  Seems like we should check `latestTime < interval` at the top of the function instead of down here.